stats/stats.go

```diff
@@ -0,0 +1,688 @@
+	if err != nil {
+		return err
+	}
+	sa.AccessManager = *accessManager
+
+	lifecycleServicesManager, err := utils.CreateLifecycleServiceManager(serverDetails, false)
+	if err != nil {
+		return err
+	}
+	sa.LifecycleServiceManager = *lifecycleServicesManager
+
+	jpdServiceManager, err := utils.CreateJPDServiceManager(serverDetails, false)
+	if err != nil {
+		return err
+	}
+	sa.JPDServicesManager = *jpdServiceManager
+
+	displayLimit = sa.DisplayLimit
+
+	sa.ServerUrl = serverDetails.Url
+
+	err = sa.GetStats()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type ArtifactoryInfo struct {
+	StorageInfo         clientUtils.StorageInfo
+	RepositoriesDetails []clientServices.RepositoryDetails `json:"-"`
+	ProjectsCount       int
+}
+
+type ProjectResources struct {
+	Resources []Resource `json:"resources"`
+}
+
+type Resource struct {
+	Type     string `json:"type"`
+	Name     string `json:"name"`
+	BinMgrID string `json:"bin_mgr_id"`
+}
+
+type AdminPrivileges struct {
+	ManageMembers   bool `json:"manage_members"`
+	ManageResources bool `json:"manage_resources"`
+	IndexResources  bool `json:"index_resources"`
+}
+
+type JPD struct {
+	ID       string    `json:"id"`
+	Name     string    `json:"name"`
+	URL      string    `json:"base_url"`
+	Status   Status    `json:"status"`
+	Local    bool      `json:"local"`
+	Services []Service `json:"services"`
+	Licenses []License `json:"licenses"`
+}
+
+func (j JPD) String() string {
+	var builder strings.Builder
+	builder.WriteString(fmt.Sprintf("Name: %s\n", j.Name))
+	builder.WriteString(fmt.Sprintf("URL: %s\n", j.URL))
+	builder.WriteString(fmt.Sprintf("Status: %s\n", j.Status.Code))
+	builder.WriteString(fmt.Sprintf("Detailed Status: %s\n", j.Status.Message))
+	builder.WriteString(fmt.Sprintf("Local: %t\n", j.Local))
+	return builder.String()
+}
+
+type Status struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+type Service struct {
+	Type   string `json:"type"`
+	Status Status `json:"status"`
+}
+
+type License struct {
+	Type       string `json:"type"`
+	Expired    bool   `json:"expired"`
+	LicensedTo string `json:"licensed_to"`
+}
+
+type RepositoryDetails struct {
+	Key         string `json:"key"`
+	Type        string `json:"type"`
+	PackageType string `json:"packageType"`
+}
+
+type ReleaseBundleResponse struct {
+	ReleaseBundles []ReleaseBundleInfo `json:"release_bundles"`
+}
+
+type ReleaseBundleInfo struct {
+	RepositoryKey     string `json:"repository_key"`
+	ReleaseBundleName string `json:"release_bundle_name"`
+	ProjectKey        string `json:"project_key"`
+}
+
+func (rb ReleaseBundleInfo) String() string {
+	var builder strings.Builder
+
+	builder.WriteString(fmt.Sprintf("ReleaseBundleName: %s\n", rb.ReleaseBundleName))
+	builder.WriteString(fmt.Sprintf("RepositoryKey: %s\n", rb.RepositoryKey))
+	builder.WriteString(fmt.Sprintf("ProjectKey: %s\n", rb.ProjectKey))
+
+	return builder.String()
```

Comment:
can we solve it using fields tags instead of a new custom string function for every struct define fields so that which has to be written can also be easily controlled.